mobius_model.h

```diff
@@ -1,1502 +0,0 @@
-
-struct mobius_model
-{
-	const char *Name;
-	const char *Version;
-	
-	entity_handle FirstUnusedEquationHandle;
-	string_map EquationNameToHandle;
-	std::vector<mobius_equation> Equations;
-	std::vector<equation_spec> EquationSpecs;
-	
-	entity_handle FirstUnusedInputHandle;
-	string_map InputNameToHandle;
-	std::vector<input_spec> InputSpecs;
-	
-	entity_handle FirstUnusedParameterHandle;
-	string_map ParameterNameToHandle;
-	std::vector<parameter_spec> ParameterSpecs;
-	
-	entity_handle FirstUnusedIndexSetHandle;
-	string_map IndexSetNameToHandle;
-	std::vector<index_set_spec> IndexSetSpecs;
-	
-	entity_handle FirstUnusedParameterGroupHandle;
-	string_map ParameterGroupNameToHandle;
-	std::vector<parameter_group_spec> ParameterGroupSpecs;
-	
-	entity_handle FirstUnusedSolverHandle;
-	string_map SolverNameToHandle;
-	std::vector<solver_spec> SolverSpecs;
-	
-	entity_handle FirstUnusedUnitHandle;
-	string_map UnitNameToHandle;
-	std::vector<unit_spec> UnitSpecs;
-	
-	std::vector<equation_batch> EquationBatches;
-	std::vector<equation_batch_group> BatchGroups;
-	
-	std::vector<mobius_preprocessing_step> PreprocessingSteps;
-	
-	timer DefinitionTimer;
-	bool Finalized;
-};
-
-#define FOR_ALL_BATCH_EQUATIONS(Batch, Do) \
-for(equation_h Equation : Batch.Equations) { Do } \
-if(Batch.Type == BatchType_Solver) { for(equation_h Equation : Batch.EquationsODE) { Do } }
-
-#if !defined(MOBIUS_EQUATION_PROFILING)
-#define MOBIUS_EQUATION_PROFILING 0
-#endif
-
-//TODO: The name "inputs" here is confusing, since there is already a different concept called input.
-//TODO: Couldn't this just be a std::vector?
-struct branch_inputs
-{
-	size_t Count;
-	index_t *Inputs;
-};
-
-struct mobius_data_set
-{
-	const mobius_model *Model;
-	
-	parameter_value *ParameterData;
-	storage_structure ParameterStorageStructure;
-	
-	double *InputData;
-	bool   *InputTimeseriesWasProvided;
-	storage_structure InputStorageStructure;
-	datetime InputDataStartDate;
-	bool InputDataHasSeparateStartDate = false; //NOTE: Whether or not a start date was provided for the input data, which is potentially different from the start date of the model run.
-	u64 InputDataTimesteps;
-	
-	double *ResultData;
-	storage_structure ResultStorageStructure;
-	
-	index_t *IndexCounts;
-	const char ***IndexNames;  // IndexNames[IndexSet.Handle][IndexNamesToHandle[IndexSet.Handle][IndexName]] == IndexName;
-	std::vector<string_map> IndexNamesToHandle;
-	bool AllIndexesHaveBeenSet;
-	
-	branch_inputs **BranchInputs; //BranchInputs[ReachIndexSet][ReachIndex] ...
-	
-	std::vector<parameter_value> FastParameterLookup;
-	std::vector<size_t> FastInputLookup;
-	std::vector<size_t> FastResultLookup;
-	std::vector<size_t> FastLastResultLookup;
-	
-	double *x0; //NOTE: Temporary storage for use by solvers
-	double *wk; //NOTE: Temporary storage for use by solvers
-
-	bool HasBeenRun;
-	u64 TimestepsLastRun;
-	
-	
-	~mobius_data_set();
-};
-
```

Comment:
Following up on the general comment about code organisation.
I suggest to move it to separate file(s).